utils/loading_cache.hh

```diff
@@ -577,6 +578,7 @@ private:
             ts_value_lru_entry& lru_entry = *_lru_list.rbegin();
             _logger.trace("shrink(): {}: dropping the entry: ms since last_read {}", lru_entry.key(), duration_cast<milliseconds>(loading_cache_clock_type::now() - lru_entry.timestamped_value().last_read()).count());
             loading_cache::destroy_ts_value(&lru_entry);
+            LoadingCacheStats::inc_privileged_on_cache_size_eviction();
```

Comment:
I noticed that we have a static_assert that inc_unprivileged_on_cache_size_eviction is noexcept (see commit 4cb245fe3c6acd029a543fdb5c922433c051271a). I don't know why this was important to do, but if it was, I think we also need one for the new privileged variant of the function.

Reply:
Sorry, I don't know that either. My best guess is that it's there for optimization purposes.
Added assertion for privileged metric.

Reply:
That's exactly the case here, thank you :) Now, that you said it, it seems to be very obvious...